src/packages/core/src/server.ts

```diff
@@ -185,6 +185,23 @@ export class Server<
       host = null;
     }
     const callbackIsFunction = typeof callback === "function";
+
+    // Method signature specifies port: number, but we parse a string if provided
+    // inspiration taken from nodejs internal port validator
+    // https://github.com/nodejs/node/blob/8c4b8b201ada6b76d5306c9c7f352e45087fb4a9/lib/internal/validators.js#L208-L219
+    if ((typeof port !== 'number' && typeof port !== 'string') ||
+      (typeof port === 'string' && String.prototype.trim.apply(port).length === 0) ||
```

Comment:
We don't need to do `String.prototype.trim.apply(port)` here. Node.js references their internal version just in case userland code has modified `trim`.  We're OK using "tampered" built-ins because it's close to impossible for us to avoid it (and sometimes user's may want to changes these built-ins in order to affect the code they are running).

Reply:
Fair enough - I was thinking something about avoiding calling .trim() on a non-string type, but we check the type right before this call.